tests/AsPipelineTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Lorisleiva\Actions\Tests;
+
+use Illuminate\Support\Facades\Pipeline;
+use Lorisleiva\Actions\Concerns\AsAction;
+use Lorisleiva\Actions\Tests\Stubs\PipelinePassable;
+
+class AsPipelineTest
+{
+    use AsAction;
+
+    public function handle(PipelinePassable $passable): void
+    {
+        $passable->increment();
+    }
+
+    public function asPipeline(PipelinePassable $passable): void
+    {
+        $this->handle($passable);
+    }
+}
+
+it('can run as a pipe in a pipeline, with an explicit asPipeline method', function () {
+    $passable = Pipeline::send(new PipelinePassable)
+        ->through([
+            AsPipelineTest::class,
+            AsPipelineTest::class,
+            AsPipelineTest::class,
+            AsPipelineTest::class,
+        ])
+        ->thenReturn();
+
+    expect(is_a($passable, PipelinePassable::class))->toBe(true);
+    expect($passable->count)->toBe(4);
+});
+
+it('can run with an arbitrary via method configured on Pipeline', function () {
+    $passable = Pipeline::send(new PipelinePassable)
+        ->via('arbitraryMethodThatDoesNotExistOnTheAction')
```

Comment:
Side note: I'm curious to understand why this works btw. Is it because if a pipeline class has an `__invoke` method, the framework will fallback to using this instead of `arbitraryMethodThatDoesNotExistOnTheAction`?